app/src/matchmaking/tournament/roundrobin/roundrobin.cpp

```diff
@@ -75,6 +73,8 @@ void RoundRobin::create() {
             // Score of Engine1 vs Engine2: 95 - 92 - 0  [0.508] 187
             // Score of Engine1 vs Engine2: 94 - 92 - 0  [0.505] 186
             std::lock_guard<std::mutex> lock(output_mutex_);
+            
+            std::string finish_message = output_->endGame(configs, stats, reason, game_id);
```

Comment:
you shouldnt need returning strings here if you move it after the mutex then there is no racy behavior